### PR TITLE
i#3080 configure libelftc: Add DRSYM_HAVE_LIBELFTC.

### LINKED ISSUE #3080: drsyms: Support configuring without libelftc
We only use it for full demangling, which isn't always necessary (use cases like #3057 can do without). No need to bring it in internally.

I'll add a drsyms configure flag (DRSYM_HAVE_LIBELFTC), which is always set for CMake builds.

### PR DESCRIPTION
**Pull request by s-kanev**
Add a drsyms configure flag, which allows non-cmake builds to build
without libelftc. Nothing changes for the cmake build, which always sets
it.

Fixes #3080

### CODE REVIEW

File: ext/drsyms/CMakeLists.txt [hunk 1]
```diff
@@ -182,6 +182,10 @@ configure_extension(drsyms_static ON)
 configure_drsyms_target(drsyms_static)
 use_DynamoRIO_extension(drsyms_static drcontainers)
 
+# we always use the elftoolchain library when building with cmake
+append_property_string(TARGET drsyms COMPILE_FLAGS "-DDRSYM_HAVE_LIBELFTC")
+append_property_string(TARGET drsyms_static COMPILE_FLAGS "-DDRSYM_HAVE_LIBELFTC")
```
Comment (derekbruening) [inline ext/drsyms/CMakeLists.txt]: Put this into configure_drsyms_target to avoid duplicating

Comment (s-kanev) [reply ext/drsyms/CMakeLists.txt]: Done.

**Comment by derekbruening** [comment]
I'm confused by the close and branch deletion?

**Comment by s-kanev** [comment]
Sorry for the bull-in-china-shop. I guess my good old "delete the fork branch and force push instead of merging" approach now closes the PR (without allowing me to reopen). Opened #3085.

**Comment by derekbruening** [comment]
No force pushes: it messes with the review history.  We use separate commits and then squash-and-merge into master.

**Pull request closed without merging**